django/bossspatialdb/views.py

```diff
@@ -488,3 +488,117 @@ def delete(self, request, collection, experiment, channel):
+            if request.query_params["iso"].lower() == "true":
+                iso = True
+            else:
+                iso = False
+        else:
+            iso = False
+
+        try:
+            request_args = {
+                "service": "cutout",
+                "collection_name": collection,
+                "experiment_name": experiment,
+                "channel_name": channel,
+                "resolution": resolution,
+                "x_args": x_range,
+                "y_args": y_range,
+                "z_args": z_range,
+                "time_args": t_range,
+            }
+            req = BossRequest(request, request_args)
+        except BossError as err:
+            return err.to_http()
+
+        # Convert to Resource
+        resource = project.BossResourceDjango(req)   
+        
+        # Get data type
+        try:
+            self.bit_depth = resource.get_bit_depth()
+            expected_data_type = resource.get_numpy_data_type()
+        except ValueError:
+            return BossHTTPError("Unsupported data type: {}".format(resource.get_data_type()), ErrorCodes.TYPE_ERROR)
+    
+        # Set a limit of CUTOUT_MAX_SIZE 
+        if is_too_large(req, self.bit_depth):
+            return BossHTTPError("Cutout overwrite is over 500MB when uncompressed. Reduce overwrite dimensions.", ErrorCodes.REQUEST_TOO_LARGE)
+    
+        # Get the shape of the requested data clear
+        if len(req.get_time()) > 2:
+            expected_shape = (len(req.get_time()), req.get_z_span(), req.get_y_span(), req.get_x_span())
+        else:
+            expected_shape = (req.get_z_span(), req.get_y_span(), req.get_x_span())
+
+        # Create a binary numpy array for overwrite with specified shape and dtype
+        black_cuboid = np.ones(expected_shape, dtype=expected_data_type)
+
+        # Get interface to SPDB cache
+        cache = SpatialDB(settings.KVIO_SETTINGS,
+                          settings.STATEIO_CONFIG,
+                          settings.OBJECTIO_CONFIG)
+
+        # Write block to cache
+        corner = (req.get_x_start(), req.get_y_start(), req.get_z_start())
+
+        try:
+            if len(black_cuboid.shape) == 4:
+                cache.write_cuboid(resource, corner, req.get_resolution(), black_cuboid, req.get_time()[0], iso=iso)
+            else:
+                cache.write_cuboid(resource, corner, req.get_resolution(),
+                                   np.expand_dims(black_cuboid, axis=0), req.get_time()[0], iso=iso)
+        except Exception as e:
+            # TODO: Eventually remove as this level of detail should not be sent to the user
+            return BossHTTPError('Error during write_cuboid: {}'.format(e), ErrorCodes.BAD_REQUEST)
+
+        # If the channel status is DOWNSAMPLED change status to NOT_DOWNSAMPLED since you just wrote data
+        channel = resource.get_channel()
+        if channel.downsample_status.upper() == "DOWNSAMPLED":
```

Comment:
@sandyhider  @movestill  Can you use the cutout interface while a channel is being downsampled? (I can dig up the answer if no one knows)

Reply:
It would probably be a good idea to add that if we aren't doing it already.

Reply:
Couldn't find a final answer, created ticket MICRONS-1639 to track this